buildSrc/src/main/kotlin/com/saveourtool/save/buildutils/spring-boot-configuration.gradle.kts

```diff
@@ -0,0 +1,94 @@
+
+plugins {
+    kotlin("kapt")
+    id("org.springframework.boot")
+    kotlin("plugin.spring")
+}
+
+extensions.getByType<KotlinJvmProjectExtension>().jvmToolchain {
+    languageVersion.set(JavaLanguageVersion.of(Versions.jdk))
+}
+
+@Suppress("GENERIC_VARIABLE_WRONG_DECLARATION")
+val libs = the<LibrariesForLibs>()
+dependencies {
+    add("implementation", platform(libs.spring.boot.dependencies))
+    add("implementation", libs.spring.boot.starter.webflux)
+    add("implementation", libs.spring.boot.starter.actuator)
+    add("implementation", libs.micrometer.registry.prometheus)  // expose prometheus metrics in actuator
+    add("implementation", libs.jackson.module.kotlin)
+    add("implementation", libs.slf4j.api)
+    add("implementation", libs.logback.core)
+    add("implementation", "io.projectreactor.kotlin:reactor-kotlin-extensions")
+
+    add("implementation", libs.springdoc.openapi.ui)
+    add("runtimeOnly", libs.springdoc.openapi.webflux.ui)
+    add("runtimeOnly", libs.springdoc.openapi.security)
+    add("runtimeOnly", libs.springdoc.openapi.kotlin)
+    add("implementation", libs.swagger.annotations)
+    add("kapt", "org.springframework.boot:spring-boot-configuration-processor:${libs.versions.spring.boot.get()}")
+
+    add("testImplementation", libs.spring.boot.starter.test)
+    add("testImplementation", libs.mockito.kotlin)
+    add("testImplementation", libs.okhttp)
+    add("testImplementation", libs.okhttp.mockwebserver)
+    add("testImplementation", libs.kotest.assertions.core)
+}
+
+configure<SpringBootExtension> {
+    buildInfo()  // configures `bootBuildInfo` task, which creates META-INF/build-info.properties file
+}
+
+tasks.withType<BootRun>().configureEach {
+    val profiles = buildString {
+        append("dev")
+        val os = DefaultNativePlatform.getCurrentOperatingSystem()
+        when {
+            os.isWindows -> append(",win")
+            os.isMacOsX -> append(",mac")
+        }
+        if (project.path.contains("save-backend")) {
+            append(",secure")
+        }
+    }
+    environment["SPRING_PROFILES_ACTIVE"] = profiles
+}
+
+tasks.named<BootBuildImage>("bootBuildImage") {
+    imageName = "ghcr.io/saveourtool/${project.name}:${project.versionForDockerImages()}"
+    environment = mapOf(
+        "BP_JVM_VERSION" to Versions.jdk,
+        "BPE_DELIM_JAVA_TOOL_OPTIONS" to " ",
+        "BPE_APPEND_JAVA_TOOL_OPTIONS" to
+                // Workaround for https://github.com/reactor/reactor-netty/issues/564
+                "-Dreactor.netty.pool.maxIdleTime=60000 -Dreactor.netty.pool.leasingStrategy=lifo " +
+                        // Override default configuration. Intended to be used on a particular environment.
+                        "-Dspring.config.additional-location=optional:file:/home/cnb/config/application.properties"
+    )
+    isVerboseLogging = true
+    val registryPassword: String? = System.getenv("GHCR_PWD")
+    isPublish = registryPassword != null
```

Comment:
maybe refactor to kotlin style with `?.let` ?

Reply:
Refactored